src/main/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchPushProcessorImpl.java

```diff
@@ -0,0 +1,340 @@
+        }
+    }
+
+    private boolean copyRawDataToArm(ExternalObjectDirectoryEntity detsExternalObjectDirectory,
+                                     ExternalObjectDirectoryEntity armExternalObjectDirectory,
+                                     String filename,
+                                     ObjectRecordStatusEntity previousStatus, UserAccountEntity userAccount,
+                                     ObjectStateRecordEntity objectStateRecord) {
+        try {
+            if (previousStatus == null
+                || ARM_RAW_DATA_FAILED.getId().equals(previousStatus.getId())
+                || ARM_INGESTION.getId().equals(previousStatus.getId())) {
+                Instant start = Instant.now();
+                log.info("ARM PERFORMANCE PUSH START for DETS EOD {} started at {}", armExternalObjectDirectory.getId(), start);
+
+                log.info("About to push raw data to ARM for DETS EOD {}", armExternalObjectDirectory.getId());
+                armDataManagementApi.copyDetsBlobDataToArm(detsExternalObjectDirectory.getExternalLocation().toString(), filename);
+                log.info("Pushed raw data to ARM for DETS EOD {}", armExternalObjectDirectory.getId());
+
+                Instant finish = Instant.now();
+                long timeElapsed = Duration.between(start, finish).toMillis();
+                log.info("ARM PERFORMANCE PUSH END for DETS EOD {} ended at {}", armExternalObjectDirectory.getId(), finish);
+                log.info("ARM PERFORMANCE PUSH ELAPSED TIME for DETS EOD {} took {} ms", armExternalObjectDirectory.getId(), timeElapsed);
+
+                armExternalObjectDirectory.setChecksum(detsExternalObjectDirectory.getChecksum());
+                armExternalObjectDirectory.setExternalLocation(UUID.randomUUID());
+                armExternalObjectDirectory.setLastModifiedBy(userAccount);
+                externalObjectDirectoryRepository.saveAndFlush(armExternalObjectDirectory);
+
+                setRawDataPushed(detsExternalObjectDirectory, objectStateRecord);
+            }
+        } catch (Exception e) {
+            String errorMessage = String.format("Error copying DETS BLOB data for file %s - %s", detsExternalObjectDirectory.getExternalLocation(),
+                                                e.getMessage());
+            log.error("Error copying DETS BLOB data for file {}", detsExternalObjectDirectory.getExternalLocation(), e);
+            updateObjectStateRecordStatus(armExternalObjectDirectory, errorMessage);
+            return false;
+        }
+
+        return true;
+    }
+
+    private void setRawDataPushed(ExternalObjectDirectoryEntity detsExternalObjectDirectory, ObjectStateRecordEntity objectStateRecord) {
+        if (nonNull(objectStateRecord)) {
+            objectStateRecord.setFlagFileTransferToDets(true);
+            objectStateRecord.setDateFileTransferToDets(currentTimeHelper.currentOffsetDateTime());
```

Comment:
the ticket says to update the below fields instead:
`flag_file_transf_toarml`
`date_file_transf_toarml`
I guess the ticket if wrong, can we get that updated if that's the case please

Reply:
the dets fields will be populated by migration we don't touch them

Reply:
I picked the wrong fields